packages/block-library/src/post-time-to-read/edit.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	AlignmentControl,
+	BlockControls,
+	store as blockEditorStore,
+	useBlockProps,
+} from '@wordpress/block-editor';
+import { store as editorStore } from '@wordpress/editor';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useEffect } from '@wordpress/element';
+import { _x, _n, __, sprintf } from '@wordpress/i18n';
+import { count as wordCount } from '@wordpress/wordcount';
+
+/**
+ * Average reading rate - based on average taken from
+ * https://irisreading.com/average-reading-speed-in-various-languages/
+ * (Characters/minute used for Chinese rather than words).
+ */
+const AVERAGE_READING_RATE = 189;
+
+function PostTimeToReadEdit( { attributes, setAttributes } ) {
+	const { textAlign, minutesToRead } = attributes;
+
+	const content = useSelect(
+		( select ) => select( editorStore ).getEditedPostAttribute( 'content' ),
+		[]
+	);
+
+	const { __unstableMarkNextChangeAsNotPersistent } =
+		useDispatch( blockEditorStore );
+
+	/*
+	 * translators: If your word count is based on single characters (e.g. East Asian characters),
+	 * enter 'characters_excluding_spaces' or 'characters_including_spaces'. Otherwise, enter 'words'.
+	 * Do not translate into your own language.
+	 */
+	const wordCountType = _x( 'words', 'Word count type. Do not translate!' );
+
+	useEffect( () => {
+		const newMinutesToRead = Math.round(
+			wordCount( content, wordCountType ) / AVERAGE_READING_RATE
+		);
+		// This is required to keep undo working and not create 2 undo steps
+		// for the content change.
+		__unstableMarkNextChangeAsNotPersistent();
+		setAttributes( {
+			minutesToRead: content ? newMinutesToRead : undefined,
+		} );
+	}, [ content ] );
+
+	let minutesToReadString = __( 'There is no content.' );
+
+	if ( minutesToRead !== undefined ) {
+		minutesToReadString =
+			minutesToRead !== 0
+				? sprintf(
+						/* translators: %d is the number of minutes the post will take to read. */
+						_n(
+							'You can read this post in %d minute.',
+							'You can read this post in %d minutes.',
```

Comment:
I think the text could be configurable, as it's unlikely to suit everyone. I'm not sure if there are any other blocks that have a way to configure text where there's a variable involved though. That seems like a challenge!

Reply:
I initially considered introducing a prefix and suffix like `core/post-terms`.

![time-to-read](https://user-images.githubusercontent.com/54422211/193535742-9a78b7ec-a8a7-4b01-bf28-5cde27074456.png)

However, a common string would not cover all cases of singular and plural, or all cases where the time required is less than a minute.

I also considered the following sidebar setting, but this is complicated 🤔

![settings](https://user-images.githubusercontent.com/54422211/193535958-efc4f907-3b88-49ff-977c-a4c19964dd76.png)

Reply:
I thought the easiest way to start might be just a prefix, and making the unit of time an implicit part of the text that can't be changed. The only confusing thing would be how internationalization works. User entered text can't be localized at the moment, but the 'minute' or 'minutes' would be. 🤔 

Lets see if we can get some assistance from a designer on this particular matter (cc @WordPress/gutenberg-design).

Reply:
Maybe the block should just print the time, then if you want a prefix/suffix you can place it in a Row alongside Paragraphs?

Reply:
> Maybe the block should just print the time, then if you want a prefix/suffix you can place it in a Row alongside Paragraphs?

Yes, I think such an approach is possible.

However, given that [the core/post-terms block has a prefix/suffix](https://github.com/WordPress/gutenberg/blob/dbd48b6f3a6085806f2bf0ec4c28b19f5e02f563/packages/block-library/src/post-terms/block.json#L20-L27), I think the current approach makes sense. Also, if this block displayed only time, adding a prefix or suffix would create multiple paragraphs. I think it should consist of a single paragraph, including prefixes and suffixes.

Reply:
Good point.

> The only confusing thing would be how internationalization works. User entered text can't be localized at the moment

If the prefix/suffix had a placeholder, would that be localized? If so I believe that would solve any issues?

Reply:
If the prefix/suffix had a placeholder, would that be localized? If so I believe that would solve any issues?

Yes, the prefix/suffix placeholder will be localized as you can see [here](https://github.com/WordPress/gutenberg/blob/fb5f3c73f3e6779ae1aa1a8e511cb8ea46fef7a3/packages/block-library/src/post-time-to-read/edit.js#L128-L129).

However, the prefix/suffix with text entered are stored as attributes and aren't localized.
Therefore, for example, if you enter prefix / suffix in English and switch to a language other than English, only the time will be localized and the correct text will not be generated.
In early versions, there was [an implementation to make all paragraphs subject to localization](https://github.com/WordPress/gutenberg/blob/689894757aedbcae4275f3037fc44ebdef9c466d/packages/block-library/src/post-time-to-read/index.php#L24-L34), without prefix/suffix. However, this would be inconvenient because the user wouldn't be able to change the text. (This may be solved by introducing filter hooks.)